app/styles/ui/_branch-select.scss

```diff
@@ -0,0 +1,73 @@
+.branch-select-component {
+  display: inline-flex;
+
+  .base-label {
+    font-weight: var(--font-weight-semibold);
+    color: var(--text-secondary-color);
+    margin: 0 var(--spacing-half);
+  }
+
+  button {
+    border: none;
+    background-color: inherit;
+    border: none;
+    padding: 0;
+    margin: 0;
+    font-style: normal;
+    font-family: var(--font-family-monospace);
+
+    .ref-component {
+      padding: 1px;
+    }
+
+    &.button-component {
+      overflow: visible;
+
+      &:hover,
+      &:focus {
+        border: none;
+        box-shadow: none;
+
+        .ref-component {
+          border-color: var(--path-segment-background-focus);
+          box-shadow: 0 0 0 1px var(--path-segment-background-focus);
+        }
+      }
+    }
+  }
+
+  .ref-component {
+    display: inline-flex;
+    align-items: center;
+  }
+
+  .branch-select-dropdown {
+    position: absolute;
+    min-height: 200px;
+    width: 365px;
+    padding: 0;
+    margin-top: 25px;
+
+    &.popover-component {
+      &::before,
+      &::after {
+        display: none;
+      }
+    }
```

Comment:
Could we handle this via a new `PopoverCaretPosition.None` enum value? (happy to wait until upcoming PRs to address this if you want)

Reply:
That reminds me that I meant to make the caret property optional (cause me providing it in the branch select component doesn't make much sense). I will go ahead and take care of that on this PR.